proto/o5/dante/v1/dead_message.proto

```diff
@@ -54,3 +71,55 @@ message Any {
   google.protobuf.Any proto = 1;
   string json = 2;
 }
+
+message DeadMessageEvent {
+  Metadata metadata = 1 [
+    (buf.validate.field).required = true,
+    (psm.v1.event_field).metadata = true
+  ];
+
+  string message_id = 2 [
+    (buf.validate.field).string.uuid = true,
+    (psm.v1.event_field).state_key = true
+  ];
+
+  DeadMessageEventType event = 3 [
+    (psm.v1.event_field).event_type = true
+  ];
+}
+
+message DeadMessageEventType {
+  oneof type {
+    Created  created  = 1;
+    Updated  updated  = 2;
+    Replayed replayed = 3;
+    Rejected rejected = 4;
+  }
+
+  message Created {
+    DeadMessageSpec spec = 1;
+  }
+
+  message Updated {
+    DeadMessageSpec spec = 1;
```

Comment:
I think the update should only contain the content and possibly destination, which could mean reducing the scope of 'DeadMessageSpec' or perhaps splitting it up

Reply:
On further thought, the 'spec' concept doesn't apply here at all. A 'Spec' is passed in through a create or update call and persists as-is to allow the API consumer to rely on the request and response having something in common, I don't think it works when the state is created by a message, so we can just skip it and write in the actual fields we want into the events, commands and state

Reply:
I actually put this here because we did have some events of needing to very weirdly edit dead messages.

We did have a few events of the messages moving packages and it meant the grpc name had to be very painfully edited.

We also had events of the queues moving, and it was a weird scenario I can't remember fully how to get there. These of course were solved by redeploys of old versions with the right queues, replaying the messages, deleting the wrongly placed ones, and then redeploying the new versions in whatever the right order was. Basically it would have been preferable to easily move them rather than have to delete messages.

I believe that leaves just a few fields, the infra id and problem as fields we maybe don't want to edit? Basically I got down to "why not?" on allowing the last few to be edited, but could absolutely see reason to not let them.